schemas/core/components/state.json

```diff
@@ -19,6 +19,12 @@
         "FINISHED"
       ]
     },
+    "bookingStateUnknown": {
+      "const": "UNKNOWN"
```

Comment:
Why not just add it as a valid booking state.
Alternatively `bookingState` can should oneOf [enum, unknown]. `maybeBookingState` is not a descriptive name

Reply:
UNKNOWN is currently only valid in stateLog. We might have some contexts where having a _broken_ value makes sense and others where we want the value to really be a _useful_ value. Of course we _could_ just add it to the list but then we can not distinguish between cases where it is ok for the value to be unknown and other cases where it should not be unknown. We don't need to fight over this but it would be nice to hear some discussion over which way others would like to go. I choose this one because it is always possible to loosen the schema later but making it stricter is impossible since we'll have old data that would need to be converted to the stricter form.

Starting variable names with maybe is a common practice for values that could either be actual values or some kind of garbage like undefined, none, null, ... Alternatively, we could use option. Both are mentioned in https://en.wikipedia.org/wiki/Option_type but I though maybeBookingState would more clear than optionBookingState. Maybe we could go with optionalBookingState if you think that sounds better.

Reply:
Value `UNKNOWN` is actually a booking state that is used to indicate corruption and not only a meta state that exists in `stateLog`. It prevents transactional rollback and fake states ( ACTIVATED but not actually activated ). Unknown signals the booking step function to stop monitoring the booking's lifecycle for further manual handling. Most Unknown bookings are handled by level 2 supports, which are then decided to whether close it or refund it

What i meant about the second point is that you can have
```
bookingState: {
   oneOf: [
     { enum: [PAID, RESERVED ...] },
     { maybeBookingState }
  ]
}
```
So ultimately the schema definition that is referenced to is `bookingState` and not `maybeBookingState`

Reply:
Oh. I'll change it then if it is a real state. I thought it was a special null value used only in state log. I wonder why the state `UNKNOWN` missing from the schema has never been a problem before.

I understood what you meant. However, it is often easier to define optional types explicitly rather than define non-optional types explicitly. See example below.

For example in TypeScript you can separate null as an after thought as follows.
```
type PhoneNumber = number | null
type RealPhoneNumber = PhoneNumber extends null ? never : PhoneNumber;
```
However the code becomes a lot simpler if you define a maybe type separately.
```
type PhoneNumber = number
type MaybePhoneNumber = phoneNumber | null
```

Reply:
I updated the PR with `UNKNOWN` being an actual state.